flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -582,7 +583,18 @@ private void startTriggeringCheckpoint(CheckpointTriggerRequest request) {
 
 							return null;
 						},
-						timer));
+						timer)
+					.exceptionally(error -> {
+						isTriggering = false;
+						if (!isShutdown()) {
+							ExceptionUtils.rethrow(error);
+						} else if (error instanceof RejectedExecutionException) {
+							LOG.debug("Execution rejected during shutdown");
+						} else {
+							LOG.warn("Error encountered during shutdown", error);
+						}
```

Comment:
Can't we collapse these two cases into logging `error` on debug if `isShutdown() == true`?

Reply:
I think we shouldn't log unexpected exceptions on `debug` level.

Reply:
In case of a shut down I would argue that exceptions can occur if the concurrent tasks are not properly stopped before. But it does not hurt to keep in on `warn` given that it should not really occur.